web3/pm.py

```diff
@@ -1,23 +1,597 @@
+from abc import (
+    ABC,
+    abstractmethod,
+)
+import json
+from typing import (
+    Any,
+    Dict,
+    Iterable,
+    List,
+    NewType,
+    Tuple,
+)
+
+from eth_utils import (
+    is_canonical_address,
+    is_checksum_address,
+    to_bytes,
+    to_canonical_address,
+    to_text,
+    to_tuple,
+)
+from eth_utils.toolz import (
+    concat,
+)
+from ethpm import (
+    ASSETS_DIR,
+    Package,
+)
+from ethpm.typing import (
+    URI,
+    Address,
+    Manifest,
+)
+from ethpm.validation import (
+    validate_package_name,
+    validate_package_version,
+)
+from pytest_ethereum.deployer import (
+    Deployer,
+)
+
+from web3 import Web3
+from web3._utils.ens import (
+    is_ens_name,
+)
+from web3.exceptions import (
+    InvalidAddress,
+    NameNotFound,
+    PMError,
+)
 from web3.module import (
     Module,
 )
 
-try:
-    from ethpm import (
-        Package,
-    )
-except ImportError as exc:
-    raise Exception(
-        "To use the (alpha) ethpm package, you must install it explicitly. "
-        "Install with `pip install --upgrade ethpm`."
-    ) from exc
+TxReceipt = NewType("TxReceipt", Dict[str, Any])
+
+
+# Package Management is still in alpha. It is not automatically available on a web3 object.
+# To use the `PM` module, attach it to your web3 object using the `attach` facility
+# PM.attach(web3, 'pm')
+
+
+class ERCRegistry(ABC):
+    """
+    The ERCRegistry class is a base class for all registry implementations to inherit from. It
+    defines the methods specified in `ERC 1319 <https://github.com/ethereum/EIPs/issues/1319>`__.
+    All of these methods are prefixed with an underscore, since they are not intended to be
+    accessed directly, but rather through the methods on ``web3.pm``.
```

Comment:
One important reason that we usually have for methods that are "not intended to be accessed directly" is that the API might change without a major version bump. Would be nice to add a note about whether these method signatures subject to change. That's not super relevant during alpha, but we might as well document for stable now.